lib/Doctrine/DBAL/Driver/PDOStatement.php

```diff
@@ -102,7 +112,13 @@ public function closeCursor()
     public function execute($params = null)
     {
         try {
-            return parent::execute($params);
+            $result = parent::execute($params);
+
+            if (! $this->connection) {
```

Comment:
This comment is invalid - the connection passed in as constructor is a design decision, and nullability is just a side-effect of BC compliance. This code should never be skipped inside DBAL, but it could be skipped if the class is used directly in userland.